packages/@orbit/record-cache/src/operators/async-patch-operators.ts

```diff
@@ -120,10 +120,16 @@ export const AsyncPatchOperators: Dict<AsyncPatchOperator> = {
 
     const relatedRecords: RecordIdentity[] =
       deepGet(record, ['relationships', relationship, 'data']) || [];
-    relatedRecords.push(relatedRecord);
+    const hasRecordInCache = !!relatedRecords.find(r =>
+      equalRecordIdentities(r, relatedRecord)
+    );
```

Comment:
There is a `recordsInclude` method already that duplicates this logic: https://github.com/orbitjs/orbit/blob/master/packages/@orbit/data/src/record.ts#L91

If we want to optimize the performance, say by using `find`, we should do so there to centralize the benefits.